Contributing to GitHub

Rnhatch opened the pull request:
After Monday's discussion about contribution obstacles,  I thought I'd draft this.  Its "Github contribution lite".   Another document is needed for fuller "local dev" oriented contribution. 

The document I've included has a placeholder image for the PR Conversation process.  I'm sure we'll get enough traffic on this PR to make a good image.  I'll also capture a few images of the final stages of the process (squash and merge and  delete branch) and include them in an update.

Enjoy.

itmightbecody commented on reports/contributing-to-github/README.md:
```diff
@@ -0,0 +1,119 @@
+**Step 7. Commit your changes early and often.**
+- Scroll to the bottom of your document 
+- Write a commit message
+- Select the `commit directoy to the <<branch name>> branch` option
```
```suggestion
- Select the `commit directly to the <<branch name>> branch` option
```

itmightbecody commented:
@Rnhatch Give this a readover to see if you agree with the changes I submitted. Also give the section on linting a proofread.

Rnhatch commented:
@itmightbecody   I really like your Linting section - but I think it actually belongs before the PR section,  and probably stands alone as a major section.   I'll make that commit, and so some proofing.

Rnhatch commented:
@itmightbecody  I adjusted my earlier comment and moved the linting section again.   I also edited it slightly for the sake of the NEWBS.  Also added some images to the squash and merge section.  I think this is ready to go..

Rnhatch commented:
The forking info looks good.   The document might be helped with some formatting and image sizing,  but I think its good enough.  If you approve,   I merge..
